.bones/tasks/marky-h6gg.md

```
@@ -0,0 +1,124 @@
+\`\`\`
+
+### 2. blob.zig readStruct/writeStruct — debug bounds assert
+**File:** zig/src/engine/blob.zig L199-210
+**Risk:** LOW — all callers use offsets from computeSectionOffsets on validated blobs. But pub functions with no bounds check.
+**Fix:** Add debug assert before memcpy. No API change (keeps void/T return types).
+
+\`\`\`zig
+pub fn writeStruct(comptime T: type, buf: []u8, offset: usize, value: T) void {
+    std.debug.assert(offset + @sizeOf(T) <= buf.len);  // bounds check (debug only)
+    const src: [*]const u8 = @ptrCast(&value);
+    @memcpy(buf[offset..][0..@sizeOf(T)], src[0..@sizeOf(T)]);
```

markdownlint-cli2 (0.21.0)

[warning] Reference links and images should use a label that is defined
Missing link or image reference definition: "0..@sizeof(t)"

(MD052, reference-links-images)